### PR TITLE
Polished Syndicate Teleporter effects

### PR DESCRIPTION
**Pull request by carlarctg**
## About The Pull Request

Altered the blood loss on teleportation. Instead of silently always losing ten blood, you now lose ~forty 25% of the time, fitting in with the teleporter's random design. In practice this usually evens out, but the immediate side-effects when it does happen will be much more noticeable. The rest of the time you'll still lose a tiny bit.

Improved the visible effects when using the teleporter. On teleporting, it causes a tiny blood red wave at the destination, which is slightly larger and more vibrant red if you lose blood. Telefragging someone now throws them a random turf away alongside the stun.

Fixed NOBLOOD check missing from non-emergency teleport bloodloss.
## Why It's Good For The Game

> Altered the blood loss on teleportation. Instead of silently always losing ten blood, you now lose ~15 25% of the time, fitting in with the teleporter's random design. In practice this usually evens out, but the immediate side-effects when it does happen will be much more noticeable. The rest of the time you'll still lose 5.

The design of the teleporter is basically gambling your life away in exchange for a powerful escape/entry device. I think it's a lot more fitting if the blood lost is more random with this in mind - you could use up all the charges without any real damage, or you could suddenly lose a byunch of. In the end the blood cost is similar but this way it'll be more visually noticeable the cost of the item.

> Improved the visible effects when using the teleporter. On teleporting, it causes a tiny blood red wave at the destination, which is slightly larger and more vibrant red if you lose blood. Telefragging someone now throws them a random turf away alongside the stun.

Basically just visual polish that I think looks cool. The throwing is becaue the effect's kinda lame.
## Changelog
:cl:
add: Altered the blood loss on teleportation. Instead of silently always losing ten blood, you now lose ~15 25% of the time, fitting in with the teleporter's random design. In practice this usually evens out, but the immediate side-effects when it does happen will be much more noticeable. The rest of the time you'll still lose 5.
qol: Improved the visible effects when using the teleporter. On teleporting, it causes a tiny blood red wave at the destination, which is slightly larger and more vibrant red if you lose blood. Telefragging someone now throws them a random turf away alongside the stun.
fix: Fixed NOBLOOD check missing from non-emergency teleport bloodloss.
/:cl:

### CODE REVIEW

**Comment by EnterTheJake** [comment]
Eeeeeeh, not sure about this one chief, experi-teleporter is already fairly dependant on RNG, one bad roll can get you gibbed. losing all of your blood in a few charges kind of sucks.

File: code/game/objects/items/teleportation.dm [hunk 1]
```diff
@@ -427,7 +429,14 @@
 		charges = max(charges - 1, 0)
 		new /obj/effect/temp_visual/teleport_abductor/syndi_teleporter(current_location)
 		new /obj/effect/temp_visual/teleport_abductor/syndi_teleporter(destination)
-		make_bloods(current_location, destination, user)
+		new /obj/effect/temp_visual/circle_wave/syndi_teleporter(destination)
+		var/blood_ejection = FALSE
+		if(!HAS_TRAIT(user, TRAIT_NOBLOOD))
+			blood_ejection = make_bloods(current_location, destination, user)
+		if(blood_ejection)
```
Comment (MrMelbert) [inline code/game/objects/items/teleportation.dm:436]: Why not use the return value of `make_bloods`, and check noblood there (if noblood return false)

File: code/game/objects/items/teleportation.dm [hunk 2]
```diff
@@ -503,8 +519,32 @@
...
+	if(prob(25) && bleed_amount)
+		playsound(src, 'sound/effects/wounds/pierce1.ogg', 40, vary = TRUE)
+		to_chat(user, span_warning("Blood visibly spurts out of your skin as [src] fails to teleport your body coherently!"))
+		carbon_user.bleed(bleed_amount * 0.9)
+		carbon_user.spray_blood(pick(GLOB.alldirs), rand(1, 3))
+		return TRUE
+	else return FALSE
+	// retval used for picking wave type
+
+/// Visual effect spawned when the bioscrambler scrambles your bio
```
Comment (MrMelbert) [inline code/game/objects/items/teleportation.dm:536]: Nice comment

File: code/game/objects/items/teleportation.dm [hunk 3]
```diff
@@ -503,8 +519,32 @@
...
+	carbon_user.bleed(bleed_amount * 0.1)
+	// sometimes lose a lot
+	// average evens out to 10 per teleport, but the randomness spices things up
+	if(prob(25) && bleed_amount)
+		playsound(src, 'sound/effects/wounds/pierce1.ogg', 40, vary = TRUE)
+		to_chat(user, span_warning("Blood visibly spurts out of your skin as [src] fails to teleport your body coherently!"))
+		carbon_user.bleed(bleed_amount * 0.9)
+		carbon_user.spray_blood(pick(GLOB.alldirs), rand(1, 3))
+		return TRUE
+	else return FALSE
```
Comment (MrMelbert) [inline code/game/objects/items/teleportation.dm:533]:
```suggestion
	return FALSE
```

**Comment by Ghommie** [comment]
I agree with EnterTheJake, the syndicate teleporter is already a high-stakes item which warrants extensive knowledge of the station design, a pair of meson googles and/or anti-bloodloss reagents, if not being a boodless humanoid like plasmamen (which have their own drawbacks btw), however these are still fairly consistent dangers that you can control; being smited by RNG in 2/3 taps meanwhile not much so, and it's frankly 'bullshit' design.

**Comment by carlarctg** [comment]
okay changed

now you always lose 5 blood

and lose 15 blood 25% of the time

File: code/game/objects/items/teleportation.dm [hunk 4]
```diff
@@ -427,7 +429,13 @@
 		charges = max(charges - 1, 0)
 		new /obj/effect/temp_visual/teleport_abductor/syndi_teleporter(current_location)
 		new /obj/effect/temp_visual/teleport_abductor/syndi_teleporter(destination)
-		make_bloods(current_location, destination, user)
+		var/blood_ejection = FALSE
+		if(!HAS_TRAIT(user, TRAIT_NOBLOOD))
+			blood_ejection = make_bloods(current_location, destination, user)
+		if(blood_ejection)
```
Comment (MrMelbert) [inline code/game/objects/items/teleportation.dm:435]:
```suggestion
		if(make_bloods(current_location, destination, user))
```

File: code/game/objects/items/teleportation.dm [hunk 5]
```diff
@@ -463,11 +471,17 @@
 		new /obj/effect/temp_visual/teleport_abductor/syndi_teleporter(mobloc)
 		new /obj/effect/temp_visual/teleport_abductor/syndi_teleporter(emergency_destination)
 		balloon_alert(user, "emergency teleport triggered!")
-		if (!HAS_TRAIT(user, TRAIT_NOBLOOD))
-			make_bloods(mobloc, emergency_destination, user)
+		var/blood_ejection = FALSE
+		if(!HAS_TRAIT(user, TRAIT_NOBLOOD))
+			blood_ejection = make_bloods(destination, emergency_destination, user)
+		if(blood_ejection)
```
Comment (MrMelbert) [inline code/game/objects/items/teleportation.dm:477]:
```suggestion
		var/blood_ejection = FALSE
		if(!HAS_TRAIT(user, TRAIT_NOBLOOD))
			blood_ejection = make_bloods(destination, emergency_destination, user)
		if(make_bloods(destination, emergency_destination, user))
```

File: code/game/objects/items/teleportation.dm [hunk 6]
```diff
@@ -503,8 +518,33 @@
 	if(!iscarbon(user))
 		return
```
Comment (MrMelbert) [inline code/game/objects/items/teleportation.dm:519]:
```suggestion
		return FALSE
```

File: code/game/objects/items/teleportation.dm [hunk 7]
```diff
@@ -495,6 +509,7 @@
 		victim.apply_damage(20, BRUTE)
 		victim.Paralyze(6 SECONDS)
 		to_chat(victim, span_warning("[user] teleports into you, knocking you to the floor with the bluespace wave!"))
+		victim.throw_at(get_step_rand(victim), 1, 1, user, spin = TRUE)
 
 ///Bleed and make blood splatters at tele start and end points
 /obj/item/syndicate_teleporter/proc/make_bloods(turf/old_location, turf/new_location, mob/living/user)
```
Comment (MrMelbert) [inline code/game/objects/items/teleportation.dm:515]:
```suggestion
/obj/item/syndicate_teleporter/proc/make_bloods(turf/old_location, turf/new_location, mob/living/user)
	if(HAS_TRAIT(user, TRAIT_NOBLOOD))
		return FALSE
```

File: code/game/objects/items/teleportation.dm [hunk 8]
```diff
@@ -503,8 +518,33 @@
...
 	var/mob/living/carbon/carbon_user = user
-	carbon_user.bleed(10)
 
+	// always lose a bit
+	carbon_user.bleed(bleed_amount * 0.25)
+	// sometimes lose a lot
+	// average evens out to 10 per teleport, but the randomness spices things up
+	if(prob(25) && bleed_amount)
+		playsound(src, 'sound/effects/wounds/pierce1.ogg', 40, vary = TRUE)
+		to_chat(user, span_warning("Blood visibly spurts out of your skin as [src] fails to teleport your body coherently!"))
```
Comment (MrMelbert) [inline code/game/objects/items/teleportation.dm:528]: Maybe should be a visible message?

**Review by MrMelbert — CHANGES_REQUESTED** [review]

**Comment by Ghommie** [comment]
Ok, blood is no longer an issue now.
